cmd/scanner/scanner.go

```diff
@@ -71,46 +91,60 @@ func getCloudProviderAccounts() ([]inventory.Account, error) {
 			account.Key("user").String(),
 			account.Key("key").String(),
 		)
-		accounts = append(accounts, *newAccount)
+		if err := s.inventory.AddAccount(*newAccount); err != nil {
+			return err
+		}
 	}
 
-	return accounts, nil
+	return nil
 }
 
 // createStockers creates and configures stocker instances for each provided account to be inventoried.
-func createStockers(accounts []inventory.Account) error {
-	for _, account := range accounts {
+func (s *Scanner) createStockers() error {
+	for i := range s.inventory.Accounts {
+		account := s.inventory.Accounts[i]
 		switch account.Provider {
 		case inventory.AWSProvider:
-			logger.Info("Adding the AWS account to be inventoried", zap.String("account", account.Name))
-			stockers = append(stockers, stocker.NewAWSStocker(account))
+			s.logger.Info("Adding the AWS account to be inventoried", zap.String("account", account.Name))
+			s.stockers = append(s.stockers, stocker.NewAWSStocker(&account, logger))
 		case inventory.GCPProvider:
-			logger.Warn("Failed to scan GCP account",
+			logger.Warn("Can't scan GCP account",
 				zap.String("account", account.Name),
 				zap.String("reason", "not implemented"),
 			)
+			// TODO: Uncomment line below when Stocker is implemented
+			//s.stockers = append(s.stockers, stocker.NewGCPStocker(&account, logger))
 		case inventory.AzureProvider:
-			logger.Warn("Failed to scan Azure account",
+			logger.Warn("Can't scan Azure account",
 				zap.String("account", account.Name),
 				zap.String("reason", "not implemented"),
 			)
+			// TODO: Uncomment line below when Stocker is implemented
+			//s.stockers = append(s.stockers, stocker.NewAzureStocker(&account, logger))
 		}
 	}
 
-	if len(stockers) == 0 {
-		return fmt.Errorf("no valid stockers created")
+	if len(s.stockers) == 0 {
+		s.logger.Warn("No accounts has been provided for scan", zap.String("credentials_file", s.credsFile))
```

Comment:
Should we return an error?
Otherwise, we continue executing the function and get the logs
```text
{"level":"warn","timestamp":"2023-09-11T09:30:11.553+0300","msg":"No accounts has been provided for scan","credentials_file":"/hots/credentials"}
{"level":"info","timestamp":"2023-09-11T09:30:11.553+0300","msg":"Saving results to the database"}
{"level":"info","timestamp":"2023-09-11T09:30:11.553+0300","msg":"Scanner finished successfully"}
```

Reply:
removed Warn and returning an error